corehq/apps/domain/auth.py

```diff
@@ -70,29 +70,29 @@ def get_username_and_password_from_request(request):
     may be null."""
     from corehq.apps.hqwebapp.utils import decode_password
 
-    if 'HTTP_AUTHORIZATION' not in request.META:
-        return None, None
-
```

Comment:
Looks like this should still stay here?
This gives an early exit if `HTTP_AUTHORIZATION` is missing otherwise `determine_authtype_from_header` will return `DIGEST` anyway and will then fail later when parsing credentials.

Reply:
There are API calls that don't set the HTTP_AUTHORIZATION header, but which are covered by `determine_authtype_from_header()`. (`determine_authtype_from_header()` will return "NONE" for uncredentialled digest and normal requests because of the `default` argument we're passing.) 

If we are looking for an early exit, we can set the first `if` to 
```python
    if auth_type == 'NONE':
        return username, password
    if auth_type == DIGEST:
        # etc.
```
but when I compared dropping two extra lines versus the speed benefit of one string comparison instead of three, I went with dropping two lines.

Reply:
@kaapstorm can you confirm this again?
From the logic [here](https://github.com/dimagi/commcare-hq/blob/37ec2c06d3ee345d66cac6d31e2ba1cb75699182/corehq/apps/domain/auth.py#L38) looks like it would default to an empty string in case `HTTP_AUTHORIZATION` header is missing and hence return the default which is digest auth.